### PR TITLE
Navigation component guidelines updates

### PR DESCRIPTION
**Pull request by mperrotti**
This is a follow-up to https://github.com/primer/design/pull/744

I've updated the guidance for UnderlineNav and SegmentedControl, and added guidance for UnderlinePanels.

More updates will be made to SegmentedControl guidance once we implement the component changes needed to support tab-like behavior.

### CODE REVIEW

**Review by joshblack — APPROVED** [review]
Love the updates! Very clear

File: content/components/segmented-control.mdx [hunk 1]
```diff
@@ -112,22 +162,22 @@ A segmented control should have 2–5 choices with text labels, or up to 6 icon-
...
-Be consistent with the way visual cues are used in each button of the segmented control. Use either all icons, all text labels, or all text labels with icons.
+Be consistent with the way visual cues are used in each segment. Use either all icons, all text labels, or all text labels with icons.
 
 Do not change icons to communicate state.
 
 <DoDontContainer>
   <Do>
     <img
-      src="https://user-images.githubusercontent.com/2313998/182705431-fe063f8f-6678-44b7-81a4-9ed3a6968a1e.png"
+      src="https://github.com/primer/design/assets/2313998/4b1bfdea-d222-4a2d-a60f-a6adfb7d33f1"
```
Comment (keithamus) [inline content/components/segmented-control.mdx:172]: Should these be updated to the new tooltip style?

Comment (langermank) [reply content/components/segmented-control.mdx:172]: @mperrotti if you want to skip updating this for now, we could open a separate PR and update all images that have tooltips. I'm happy to tackle that on some future slow day

Comment (mperrotti) [reply content/components/segmented-control.mdx:172]: Good catch @keithamus 

Thanks @langermank! Yea, let's do that 🙂

**Pull request merged**